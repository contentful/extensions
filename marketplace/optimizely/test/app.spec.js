import React from 'react';
import { cleanup, render } from '@testing-library/react';

import App from '../src/app';
import mockProps from './mockProps';

describe('App', () => {
  afterEach(cleanup);

  it('should match the snapshot', () => {
    const { container } = render(<App {...mockProps} />);

    expect(container).toMatchSnapshot();
  });
});
