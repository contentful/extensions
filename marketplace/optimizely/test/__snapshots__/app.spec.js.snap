// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div>
  <div
    class="css-175t7kh"
  >
    <div
      class="css-3p38p2"
    >
      <div
        class="css-xieh1a"
      >
        <svg
          class="Icon__Icon___38Epv css-uoder Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
          />
        </svg>
        <span>
          Select experiment
        </span>
      </div>
      <svg
        class="Icon__Icon___38Epv css-13af3fo Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
        data-test-id="cf-ui-icon"
        height="18"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
      <div
        class="css-xieh1a"
      >
        <svg
          class="Icon__Icon___38Epv css-uoder Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
          />
        </svg>
        <span>
          Add content
        </span>
      </div>
      <svg
        class="Icon__Icon___38Epv css-13af3fo Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
        data-test-id="cf-ui-icon"
        height="18"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
      <div
        class="css-xieh1a"
      >
        <svg
          class="Icon__Icon___38Epv css-uoder Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
          />
        </svg>
        <span>
          Publish variations
        </span>
      </div>
      <svg
        class="Icon__Icon___38Epv css-13af3fo Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
        data-test-id="cf-ui-icon"
        height="18"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
      <div
        class="css-xieh1a"
      >
        <svg
          class="Icon__Icon___38Epv css-uoder Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
          />
        </svg>
        <span>
          Start experiment
        </span>
      </div>
    </div>
    <hr
      class="css-14380qe"
    />
    <h2
      class="Heading__Heading___83D3K css-1fobf8d"
      data-test-id="cf-ui-heading"
    >
      Referenced in:
    </h2>
    <div
      class="css-1y5g446"
    >
      <svg
        aria-label="Loading component..."
        class="SkeletonContainer__SkeletonContainer___23jiu"
        data-test-id="cf-ui-skeleton-form"
        height="30px"
        preserveAspectRatio="xMidYMid meet"
        role="img"
        width="100%"
      >
        <title>
          Loading component...
        </title>
        <rect
          clip-path="url(#references-section)"
          height="100%"
          style="fill: url(#cf-ui-skeleton-clip-gradient);"
          width="100%"
          x="0"
          y="0"
        />
        <defs>
          <clippath
            id="references-section"
          >
            <rect
              height="16"
              rx="0"
              ry="0"
              width="100%"
              x="0"
              y="0"
            />
          </clippath>
          <lineargradient
            id="cf-ui-skeleton-clip-gradient"
          >
            <stop
              offset="0%"
              stop-color="#e5ebed"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                dur="2s"
                id="animation1"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#f7f9fa"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                begin="animation1.begin+0.25s"
                dur="2s"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#e5ebed"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                begin="animation1.begin+0.5s"
                dur="2s"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
    </div>
    <hr
      class="css-14380qe"
    />
    <h2
      class="Heading__Heading___83D3K css-1fobf8d"
      data-test-id="cf-ui-heading"
    >
      Experiment:
    </h2>
    <div
      class="SelectField__SelectField___kbQlf"
      data-test-id="cf-ui-select-field"
    >
      <div
        class="SelectField__SelectField__label-wrapper___3jGwo"
      >
        <label
          class="FormLabel__FormLabel___3d6zQ"
          data-test-id="cf-ui-form-label"
          for="experiment"
        >
          Optimizely experiment
          <span
            class="FormLabel__FormLabel__required-text___3mvdm"
          >
            (
            required
            )
          </span>
        </label>
      </div>
      <div
        class="Select__Select__wrapper___2pBIz Select__Select--large___1PgJW"
      >
        <select
          aria-label="experiment"
          class="Select__Select___31Z46 a11y__focus-border--default___60AXp Select__Select--disabled___3n58N"
          data-test-id="cf-ui-select"
          disabled=""
          id="experiment"
          name="experiment"
          required=""
        >
          <option
            data-test-id="cf-ui-select-option"
            value="-1"
          >
            Fetching experiments...
          </option>
        </select>
        <svg
          class="Icon__Icon___38Epv Select__Select__icon___boRAZ Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
          data-test-id="cf-ui-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </div>
    </div>
    <hr
      class="css-14380qe"
    />
    <h2
      class="Heading__Heading___83D3K"
      data-test-id="cf-ui-heading"
    >
      Variations:
    </h2>
    <div
      class="css-1pfvl5o"
    >
      <svg
        aria-label="Loading component..."
        class="SkeletonContainer__SkeletonContainer___23jiu"
        data-test-id="cf-ui-skeleton-form"
        height="100"
        preserveAspectRatio="xMidYMid meet"
        role="img"
        width="100%"
      >
        <title>
          Loading component...
        </title>
        <rect
          clip-path="url(#variations-section)"
          height="100%"
          style="fill: url(#cf-ui-skeleton-clip-gradient);"
          width="100%"
          x="0"
          y="0"
        />
        <defs>
          <clippath
            id="variations-section"
          >
            <rect
              height="16"
              rx="0"
              ry="0"
              width="100%"
              x="0"
              y="20"
            />
            <rect
              height="16"
              rx="0"
              ry="0"
              width="100%"
              x="0"
              y="44"
            />
            <rect
              height="16"
              rx="0"
              ry="0"
              width="80%"
              x="0"
              y="68"
            />
          </clippath>
          <lineargradient
            id="cf-ui-skeleton-clip-gradient"
          >
            <stop
              offset="0%"
              stop-color="#e5ebed"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                dur="2s"
                id="animation1"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#f7f9fa"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                begin="animation1.begin+0.25s"
                dur="2s"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#e5ebed"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                begin="animation1.begin+0.5s"
                dur="2s"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
    </div>
  </div>
</div>
`;
